client/hwlib/src/collectors/hardware_info.rs

```diff
@@ -182,26 +186,42 @@ pub(crate) fn table_load_from_device(
     table_file: &Path,
 ) -> Result<SMBiosData, anyhow::Error> {
     let version = SMBiosEntryPoint64::try_load_from_file(entry_file)
+        .with_context(|| {
+            format!(
+                "Failed to load SMBIOS entry point 64 from: {}",
+                entry_file.display()
+            )
+        })
         .map(|entry_point| SMBiosVersion {
             major: entry_point.major_version(),
             minor: entry_point.minor_version(),
             revision: 0,
         })
         .or_else(|err| {
-            if err.kind() != ErrorKind::InvalidData {
-                return Err(err);
+            // Downcast to std::io::Error to check the kind
+            if let Some(io_err) = err.downcast_ref::<IoError>() {
+                if io_err.kind() != ErrorKind::InvalidData {
+                    return Err(err);
+                }
+                SMBiosEntryPoint32::try_load_from_file(entry_file)
+                    .with_context(|| {
+                        format!(
+                            "Failed to load SMBIOS entry point 32 from: {}",
```

Comment:
Perhaps I lack context (pun intended), but is entry-point 32 an obvious concept from the outside? Is there another way to describe this for mere mortals? :P